integration/src/integration/java/pl/allegro/tech/hermes/integration/JmsConsumingTest.java

```diff
@@ -64,6 +65,43 @@ public void shouldPublishAndConsumeJMSMessageWithTraceId() throws Exception {
         assertThat(jmsEndpoint.waitAndGetLastMessage()).assertStringProperty("TraceId", traceId);
     }
 
+    @Test
+    public void shouldPublishAndConsumeJmsMessageWithTraceHeaders() throws Exception {
+
+        // given
+        String message = "{\"hello\": \"world\"}";
+        String traceId = UUID.randomUUID().toString();
+        String spanId = UUID.randomUUID().toString();
+        String parentSpanId = UUID.randomUUID().toString();
+        String sampled = "1";
+        String reported = "0";
+
+        // and
+        Topic topic = operations.buildTopic("publishJmsGroupWithTrace", "topic");
+        operations.createSubscription(topic, "subscription", jmsEndpointAddress(JMS_TOPIC_NAME));
+        WebTarget client = ClientBuilder.newClient().target(FRONTEND_URL).path("topics").path(topic.getQualifiedName());
+        jmsEndpoint.expectMessages(TestMessage.of("hello", "world"));
+
+        // when
+        Response response = client
+                .request()
+                .header("Trace-Id", traceId)
+                .header("Span-Id", spanId)
+                .header("Parent-Span-Id", parentSpanId)
+                .header("Trace-Sampled", sampled)
+                .header("Trace-Reported", reported)
+                .post(Entity.entity(message, MediaType.APPLICATION_JSON));
+
+        // then
+        assertThat(response).hasStatus(Response.Status.CREATED);
+        Message lastMessage = jmsEndpoint.waitAndGetLastMessage();
+        assertThat(lastMessage).assertStringProperty("TraceId", traceId);
+        assertThat(lastMessage).assertStringProperty("SpanId", spanId);
+        assertThat(lastMessage).assertStringProperty("ParentSpanId", parentSpanId);
+        assertThat(lastMessage).assertStringProperty("TraceSampled", sampled);
+        assertThat(lastMessage).assertStringProperty("TraceReported", reported);
```

Comment:
Ok, I understand. From what I remember we have very little JMS subscribers. If we check how many of them adopted these tracing features then maybe it would be possible to use underscores as you suggested.